src/RazorSlices/RazorSlice.ResolveAndCreate.cs

```diff
@@ -281,6 +443,11 @@ private static SliceFactory<TModel> ResolveSliceFactoryImpl<TModel>(string slice
 
         var sliceDefinition = _slicesByName[sliceName];
 
+        if (sliceDefinition.InjectableProperties.Any())
```

Comment:
Let's avoid the ambiguity of the cost of the `Any()` call here and instead update `SliceDefinition` with a `bool HasInjectableServices` flag that makes this check a simple property value check during slice creation.